cmd/catchpointdump/file.go

```diff
@@ -137,16 +173,54 @@ func loadCatchpointIntoDatabase(ctx context.Context, catchupAccessor ledger.Catc
 			// we already know it's valid, since we validated that above.
 			protocol.Decode(balancesBlockBytes, &fileHeader)
 		}
+		if time.Now().Sub(lastProgressUpdate) > 50*time.Millisecond && len(fileBytes) > 0 {
+			lastProgressUpdate = time.Now()
+			printLoadCatchpointProgressLine(int(float64(progress)*50.0/float64(len(fileBytes))), 50, int64(progress))
+		}
 	}
 }
 
+func printDumpingCatchpointProgressLine(progress int, barLength int, dld int64) {
```

Comment:
looks like it is different from the `printLoadCatchpointProgressLine` func only in "Loaded" vs "Dumped" strings constants, consider generalizing

Reply:
I'll leave that refactoring for now; I have extracted most of the content outside so that the content is quite "shallow". We can create "unified" progress bars in a separate ticket.